src/Tomlyn/Parsing/Lexer.cs

```diff
@@ -525,7 +525,7 @@ private void ReadNumberOrDate(char32? signPrefix = null, TextPosition? signPrefi
                     previousCharIsDigit = _c == '0';
                     if (previousCharIsDigit)
                     {
-                        _textBuilder.Append(_c);
+                        _textBuilder.Append(_c.ToString());
```

Comment:
Not sure to understand the fix. Why ToString() is fixing it here?

Reply:
Because `_c` is a `char32` which has an implicit operator conversion to `int` so the wrong overload of `_textBuilder.Append` is chosen an we end up with `048:…` instead of `00:…`.

You can try to run the `TestLocalTime` test introduced in a0fa54238cfb09905cce5c538ee97c060675e230 and you'll see. 😉